durable-storage/src/database.rs

```diff
@@ -92,18 +94,19 @@ impl Database {
     ///  - The offset is larger than the length of the associated value.
     pub(crate) fn write(
         &mut self,
-        _key: &Key,
+        key: Key,
         offset: usize,
-        _data: &[u8],
+        data: Bytes,
     ) -> Result<usize, DatabaseError> {
         if offset != 0 {
             let value = None;
             // TODO : Implement [`MerkleLayer::node::get_mut`] in RV-827
             value.ok_or(DatabaseError::KeyNotFound)?
         } else {
-            // TODO: Implement database writes in RV-827
-            //     : Compare the offset and the size of the value in RV-827
-            Err(DatabaseError::OffsetTooLarge)
+            let written = data.len();
+            self.persistent.set(&key, data.clone())?;
+            self.merkle.set(key.clone(), data);
```

Comment:
I think this works without cloning.

```suggestion
            self.merkle.set(key, data);
```

Reply:
[Done](https://github.com/tezos/riscv-pvm/compare/66702f451caed81c721fb768275a7cf7feee7276..57056749eca5d7fe351dc0d610347bd9cf49a204) thanks